flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -100,6 +102,9 @@
 	 */
 	private JobID lastJobID;
 
+	/** Switch for blocking/detached job submission of the client */
+	private boolean detachedJobSubmission = false;
```

Comment:
Yes, this flag is actually just an indicator that can be set by the cluster client. Users may still submit jobs blocking or detached. I know that's not ideal, I'll try to see if we can get rid of this. I think we could get rid of the detached/blocking methods and make this the only point where the user controls the job submission mode.